packages/create-schemas/src/plugins/types-plugin.ts

```diff
@@ -0,0 +1,98 @@
+import assert from "node:assert";
+import { astToString, stringToAST } from "openapi-typescript";
+import ts from "typescript";
+import type { Plugin } from "./plugin.ts";
+import { openapiTypeScriptId } from "./openapi-typescript-plugin.ts";
+
+const componentsIdentifier = "components";
+const schemasIdentifier = "schemas";
+
+export const RESERVED_IDENTIFIERS = new Set(["paths", "webhooks", componentsIdentifier, "$defs", "operations"]);
+
+export function typesPlugin(): Plugin {
+    return {
+        name: "internal:types-plugin",
+        async transform({ id, code }) {
+            if (id !== openapiTypeScriptId) {
+                return;
+            }
+
+            const ast = stringToAST(code) as ts.Node[];
+
+            const componentsDeclaration = ast.find(isComponentsInterfaceDeclaration);
+
+            assert(componentsDeclaration, "Missing components declaration");
+            const schema = componentsDeclaration.members.find(isComponentsSchema);
+            assert(schema, "Missing components declaration");
+            assert(schema.type && ts.isTypeLiteralNode(schema.type), "Invalid schema type");
+        
+            const typeNodes = schema.type.members
+                .map(member => member.name)
+                .filter(name => name !== undefined)
```

Comment:
just to be sure, we do prune the cases where name is empty (which also shouldn't happen ever) but given we have a charAt(0) later, its a good to have

Reply:
oooh, I didn't consider empty names. I'll add it to the list of things to look out for.